torch/csrc/jit/passes/autocast.cpp

```diff
@@ -414,6 +413,15 @@ void handleBlock(Block* block, AutocastContext initial_state) {
         }
         break;
 
+      // cast softmax to fp32 only on GPU
+      case aten::softmax:
+        if (!node->schema().is_mutable() && !hasExplicitDtypeArgument(node)) {
+          auto context = current_state();
+          context.cpu_enabled = false;
+          castTensorInputs(node, aten::_autocast_to_full_precision, context);
```

Comment:
isnt the modification of `cpu_enabled` going to persist pass this op?

Reply:
I think since I'm copying the state to `auto context` (i.e. not by reference) it's not going to affect other ops